lib/tasks/import_new_redlist_data.rake

```diff
@@ -0,0 +1,158 @@
+require 'csv'
+
+namespace :import do
+
+  desc "import RedList species data into database"
+  task :new_redlist_data, [:geo_type] => [:environment] do |t, args|
+    if args[:geo_type].present?
+      send("import_for_#{args[:geo_type]}")
+    else
+      import_for_countries
+      import_for_regions
+    end
+  end
+
+  desc "import habitats occurrences"
+  task :occurrences => [:environment] do
+    species_filename = "lib/data/iucn_redlist/country_species.csv".freeze
+
+    CSV.read(species_filename, headers: true)
+       .uniq { |r| r.values_at('iso3', 'habitat') }.each do |row|
+
+      habitat = Habitat.where('LOWER(title) = ?', row['habitat'].downcase).first
+      next unless log_habitat(row, habitat)
+
+      import_occurrences(row, habitat)
+    end
+  end
+
+  def import_for_countries
+    species_filename = "lib/data/iucn_redlist/country_species.csv".freeze
+    prev_iso, prev_hab = nil, nil
+    @failed_report = []
+
+    CSV.foreach(species_filename, headers: true) do |row|
+      habitat = Habitat.where('LOWER(title) = ?', row['habitat'].downcase).first
+      next unless log_habitat(row, habitat)
+
+      import_species(row, habitat)
+      import_countries_species(row, habitat)
+
+      current_iso, current_hab = row['iso3'], row['habitat']
+      next if (current_iso == prev_iso) && (current_hab == prev_hab)
+      import_occurrences(row, habitat)
+      prev_iso, prev_hab = current_iso, current_hab
+    end
+
+    log_failed('countries')
+  end
+
+  def import_for_regions
+    species_filename = "lib/data/iucn_redlist/region_species.csv".freeze
+    @failed_report = []
+
+    CSV.foreach(species_filename, headers: true) do |row|
+      habitat = Habitat.where('LOWER(title) = ?', row['habitat'].downcase).first
+      next unless log_habitat(row, habitat)
+
+      import_species(row, habitat)
+      import_regions_species(row, habitat)
+    end
+
+    log_failed('regions')
+  end
+
+  def import_species(row, habitat)
+    species_header = Species.new.attributes.keys.freeze
+    scientific_name = row['scientific_name']
+
+    return if scientific_name == 'NA'
+    return if Species.find_by(scientific_name: scientific_name, habitat_id: habitat.id).present?
+
+    row['url'] = 'https://'.concat(row['url']) if (row['url'] != 'N/A' && !row['url'].match(/^https/))
+    row_hash = row.to_hash.slice(*species_header)
+    species = Species.new(row_hash.merge(habitat_id: habitat.id))
+    report_failed(row, species.errors.messages) unless species.save
+  end
+
+  def import_countries_species(row, habitat)
+    iso3 = row['iso3']
+
```

Comment:
I would add a check here like
 `return if scientific_name == 'NA'` just to avoid having many rows in the failed report like 
```ABW,Cold-water corals,NA,NA,NA,NA,NA,1,NA,"Species, habitat pair not found!"```

Reply:
My original intention was to report every single row that failed, so it's easier to understand what happened also by counting the number of records successfully imported and the ones that failed. 
For example, I spent quite a bit of time understanding why the number of imported ones was different; with such a report it was easier.

Hope that makes sense